### DISCUSSION TITLE
What if you think about native web components ?

### DISCUSSION THREAD

**Discussion by kaesar** [vuejs/petite-vue#113] [Ideas]
I noticed that **petite-vue** works without Virtual-DOM and I understand that there is no compiler yet, but it is great by its tiny size...
> **What if you think about native web components**?  
> Future proof thanks to the JavaScript module syntax?

That's what libraries like [**RiotJS**](https://riot.js.org) and [**Lit 2**](https://lit.dev) (by Google now) have, but these are not **Vue**.
So, imagine you don't need to import an script tag for **petit-vue** with CDN (or local), because it would be native (just the component).

> I don't know if is possible merge a component with the library.

**Comment by niutech** [comment]
In my opinion, this is not how Vue.js works and web components will certainly break compatibility with existing Vue.js code (which Petite Vue is a subset of).
Furthermore, it would require a build step to create a JS module with a web component, like Riot.js and Lit do, whereas Petite Vue is so simple that it requires just a single script from the CDN - and there is nothing wrong with it - zero build step, voila!
And the most important point: how about progressive enhancement with web components? I don't think Riot.js and Lit are progressively enhanced like Petite Vue is. They use JS strings to store HTML templates, unlike Petite Vue (and Alpine.js), which use the DOM.
What would you like to achieve by converting Petite Vue to web components?

**Comment by kaesar** [comment]
Thanks a lot @niutech , I understand your point and good questions, just let me answer you the last one.
About what would I like to achieve, I think...

1. Perhaps future proof thanks to the JavaScript module syntax
2. To get involve this technology as an standard and it can works with technologies that use native web components (e.g. [**Astro**](https://astro.build), etc.)
3. To use it with others libraries like React, so you can create core components with petite-vue (I don't know if it works right now)
4. Perhaps to bet in petite-vue for the UI in a programming language that I planned, it is called [**ABCode**](https://onmind.co/doc/code/en/ABCode.md)

Please, may be you can clarify to me some points (1, 2, 3) and I appreciate that (some times all is about to get better documentation or FAQ)

**Reply by niutech** [reply]
You're welcome!
Ad 1. What does  it mean _future proof thanks to the JavaScript module syntax_? Petite Vue can already be used as an [ES module](https://github.com/vuejs/petite-vue#manual-init). And JS global objects (like `PetiteVue`) are not going anywhere, so they will be supported for the next years to come.
That said, you can wrap Petite Vue in a Custom Element similar to [Vue.js Web Component Wrapper](https://github.com/vuejs/vue-web-component-wrapper).
